### ISSUE TITLE
Comments in the middle of robot file is caught with rule 1007 (line is under-intented)

### ISSUE THREAD

**Issue by sampeson** [MarketSquare/robotframework-robocop#332]
When putting comment to middle of file between test cases or keywords there robocop complains about intent. See attached file for example. It should not complain about it or the empty line after it.

See example file:
[comment_example.robot.zip](https://github.com/MarketSquare/robotframework-robocop/files/6264624/comment_example.robot.zip)

Exptected behavior is to accept it as it is common to have comments like this in test data.

**Comment by bhirsz** [comment]
I agree. I've prepared sample test data:
```*** Test Cases ***
# valid comment
  # invalid comment
Test1
    Keyword
    # comment
  # invalid comment

# comment
    # now also invalid

Test2
    Keyword
    FOR    ${var}  IN   1  2
        Keyword
# invalid comment
    END
```
after the fix Robocop should report only on "invalid comment".

**Comment by sampeson** [comment]
One special case still for the acceptance test, does it approve comments in the end of line, e.g.
```
*** Test Cases ***
Test
    Log      hello world   # just commenting this line
```

**Comment by bhirsz** [comment]
Yes it's fine - but to be sure I've added it to our test data for uneven/bad indents.
